docs/start-date-computation.md

```diff
@@ -0,0 +1,74 @@
+
+In this chapter, we learn how to compute start dates.
+
+### Context
+
+To be computed, a start date needs a context. here is the context we are going to work with:
+
+First we need a cohort spec. Let's assume that the cohort spec is
+
+```
+{
+    "cohortName":"GW2024",
+    "brazeCampaignName":"SV_GW_PriceRise2024",
+    "importStartDate":"2024-02-01",
+    "earliestPriceMigrationStartDate":"2024-05-20" 
+}
+```
+
+The only relevant bit of cohort spec we need for the computation of a subscription's start date is the `earliestPriceMigrationStartDate`. 
+
+Let us assume thay the notification period for this migration is `[-49, -36]`. This is Pascal's notation for the fact that we start notifying at -49 days and alarm at -36.
+
+Let us assume that our subscription is a monthly subscription paying on the 27th of each month, and let us assume that it was created on 8th July 2023.
+
+Let us assume that for this migration, marketing decided that the spread period would be 3 months (see section below for the definition of the spread period).
+
+And last, but not least, let's assume that today is 07th March 2024.
+
+### A note on anniversaries.
+
+Imagine that you work for HR and are given the task of giving a gift of the thousands of people working for a company. You can only give to somebody their gift on their birthday. It's going to take you an entire year to do so. If today is the 7th of March 2024, then you will give gifts to a bunch of people today (all the people with a 7th of March birthday) and you will finish on the 6th of March 2025.
+
+If you are then told to not use the yearly birthday, but simply the number of their birthdate, so for instance if somebody is born on Dec 23rd, you can give them their gift on 23rd of March. Then, it will take you one month to give all the gifts away. You will start today on 7th of March and will finish on 6th of April.
+
+But then you realise that if you use the monthly anniversary of the person, then there are many more people to give a gift to per day (exactly 12 times more people actually), and you can't run that fast. So you decide of a spread period. You decide that you will do it over three months. So you attribute a random number from the set { 0, 1, 2 } to each person on your distribution spreadsheet, and if the number is 0 you give them the gift the next available day number, otherwise you postpone by that random choice number of months.
+
+### Spread periods
+
+When Marketing says that they want to spead the monthlies over 3 months, they mean that the process of upgrading the monthlies to a new price (which should otherwise only need 1 month) should be planned to take 3 months. You know that there will be a random choice made as part of the decision of the subscription start date. 
+
+We could, but we never use a spread period for other billing frequencies (quarterly, annual, etc)
```

Comment:
presumably because they already have a "natural" spread through the year?

Reply:
Totally yes!